Chess-FromZeroToHero.DataAccess/Helpers/QueryableExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Chess_FromZeroToHero.DataAccess.Pagination;
+using Microsoft.EntityFrameworkCore;
+
+namespace Chess_FromZeroToHero.Contracts.Helpers
+{
+    public static class QueryableExtensions
+    {
+        public static async Task<List<T>> PaginateAsync<T>(this IQueryable<T> query, PaginationParams paginationParams)
+        {
+            int elements = await query.CountAsync();
+            int skip = (paginationParams.Page - 1) * paginationParams.ItemsPerPage;
+            int take = paginationParams.Page * paginationParams.ItemsPerPage > elements
```

Comment:
You don't need to calculate the `take` just use the `paginationParams.ItemsPerPage`. E.g.:
```
await query.Skip(skip).Take(paginationParams.ItemsPerPage).ToListAsync()
```